src/utils/AssayGeneratorMetabo.cpp

```diff
@@ -591,8 +601,34 @@ class TOPPAssayGeneratorMetabo :
     // use MRMAssay methods for filtering
     MRMAssay assay;
 
-    // filter: min/max transitions
-    assay.detectingTransitionsCompound(t_exp, min_transitions, max_transitions);
+    // sort by highest intensity - filter:  min/max transitions (targets), filter: max transitions (decoys)
```

Comment:
Have you thought about filtering before you resolve? In case you resolve for something that you would filter anyway (then you could have used the second best).

Reply:
Yes, the problem here is that the two data structures used are really different and I think it does not make sense to parse to MRMAssay and then back for the resolution and then to MRMAssay again.